zebra-network/src/peer_set/set.rs

```diff
@@ -152,6 +161,31 @@ where
         });
     }
 
+    fn check_for_background_errors(&mut self, cx: &mut Context) -> Result<(), BoxedStdError> {
+        if self.guards.is_empty() {
+            match self.handle_rx.try_recv() {
+                Ok(handles) => {
+                    for handle in handles {
+                        self.push_join_handle(handle);
```

Comment:
is there any reason this can't be just
```suggestion
                        self.guards.push(handle);
```
? it looks like the push_join_handle method is only ever called here.